### ISSUE TITLE
Windows: Wrong default config directory

### ISSUE THREAD

**Issue by zimbatm** [direnv/direnv#120]
In the `xdg.go` module we set `$HOME/.config/direnv` to be the default folder under which direnv stores it's allow files (and also looks for the direnvrc file). Under Windows the default should probably be `%APPDATA%\direnv`. I am trying to follow the [XDG Base Directory Specification](http://standards.freedesktop.org/basedir-spec/basedir-spec-latest.html).

One solution would be to create a `xdg_windows.go` file which contains a different default folder than `xdg_unix.go`. Build conditionals can be used to compile one file or the other so that xdg.go can refer to the defaultConfigDir(env) function.

Here is another app doing something similar: http://zim-wiki.org/manual/Help/Config_Files.html

**Comment by wilmoore** [comment]
For the same, the following package is commonly used in NodeJS programs:
https://www.npmjs.com/package/user-home

**Comment by mdekstrand** [comment]
@zimbatm Yes, it's my understanding as well that `%APPDATA%` is the appropriate location. Possibly the local version (instead of roaming), depending on whether the files direnv stores in this directory are intended to be portable across machines with roaming profiles.

**Comment by zimbatm** [comment]
I don't have a windows environment available but if you want to submit a PR I would be interested. Go has this nice feature where files are only compiled for the desired targets.
